panic on sendfrom RPC

Running current master (f5b98d7c5d089fe5e0ba47bb44bc55aaad3500e8).

Reproducible when running 3x `sendfrom` RPCs in very rapid succession. No panic when I slowed down and fired one at time.

```
2020-04-27 12:06:34.087 [INF] RPCS: RPC method sendfrom invoked by 127.0.0.1:40322
2020-04-27 12:06:34.087 [INF] WLLT: Returning address (account=3 branch=1 child=1)
2020-04-27 12:06:34.090 [INF] WLLT: Inserting unconfirmed transaction 05fa0f81b982735900cfa5c00eb868568210cdd7c6c4e0825c1d147f8fcb2910
panic: runtime error: slice bounds out of range [128:0]

goroutine 413515 [running]:
decred.org/dcrwallet/wallet/udb.extractBlockHeaderHeight(...)
	/home/jholdstock/code/decred/dcrwallet/wallet/udb/txmined.go:488
decred.org/dcrwallet/wallet/udb.(*Store).AddCredit(0xc00000e880, 0xfb54a0, 0xc0009cc3c0, 0xc0000c0500, 0xc00020e370, 0x100000000, 0xc000000003, 0x0, 0x0)
	/home/jholdstock/code/decred/dcrwallet/wallet/udb/txmined.go:1293 +0x6dd
decred.org/dcrwallet/wallet.(*Wallet).processTransactionRecord(0xc00063e160, 0xfac0c0, 0xc00012d3b0, 0xfb06a0, 0xc000616250, 0xc0000c0500, 0x0, 0xc00020e370, 0x0, 0x0, ...)
	/home/jholdstock/code/decred/dcrwallet/wallet/chainntfns.go:667 +0x1f89
decred.org/dcrwallet/wallet.(*Wallet).AddTransaction.func1(0xfb06a0, 0xc000616250, 0xd17f52, 0xe)
	/home/jholdstock/code/decred/dcrwallet/wallet/chainntfns.go:350 +0x1bf
decred.org/dcrwallet/wallet/walletdb.Update(0xfac0c0, 0xc00012d3b0, 0xfac2c0, 0xc000672600, 0xc0011d1d78, 0x0, 0x0)
	/home/jholdstock/code/decred/dcrwallet/wallet/walletdb/interface.go:249 +0x1a5
decred.org/dcrwallet/wallet.(*Wallet).AddTransaction(0xc00063e160, 0xfac0c0, 0xc00012d3b0, 0xc00020e2d0, 0x0, 0x0, 0xc001b66540)
	/home/jholdstock/code/decred/dcrwallet/wallet/chainntfns.go:312 +0x1b6
decred.org/dcrwallet/chain.(*Syncer).relevantTxAccepted(0xc00066caf0, 0xfac0c0, 0xc00012d3b0, 0xc0009cb000, 0x722, 0x800, 0xc001b66540, 0xbbf6a0)
	/home/jholdstock/code/decred/dcrwallet/chain/sync.go:652 +0x98
decred.org/dcrwallet/chain.(*notifier).Notify(0xc00012d0b0, 0xc0012de880, 0x12, 0xc0009cb000, 0x722, 0x800, 0x0, 0x0)
	/home/jholdstock/code/decred/dcrwallet/chain/sync.go:559 +0x7aa
github.com/jrick/wsrpc/v2.(*Client).in.func2(0xc00061c080, 0xc000612020, 0xc0000a8af0, 0xfac0c0, 0xc000610bd0, 0xc000042660, 0xc00013a1e0)
	/home/jholdstock/code/gopath/pkg/mod/github.com/jrick/wsrpc/v2@v2.3.2/rpc.go:388 +0x1a3
created by github.com/jrick/wsrpc/v2.(*Client).in
	/home/jholdstock/code/gopath/pkg/mod/github.com/jrick/wsrpc/v2@v2.3.2/rpc.go:381 +0x3df

```

Yes

```
2020-04-28 10:40:31.628 [INF] RPCS: RPC method sendfrom invoked by 127.0.0.1:49860
2020-04-28 10:40:31.628 [INF] RPCS: RPC method sendfrom invoked by 127.0.0.1:49860
2020-04-28 10:40:31.628 [INF] RPCS: RPC method sendfrom invoked by 127.0.0.1:49860
2020-04-28 10:40:31.628 [INF] RPCS: RPC method sendfrom invoked by 127.0.0.1:49860
2020-04-28 10:40:31.628 [INF] RPCS: RPC method sendfrom invoked by 127.0.0.1:49860
2020-04-28 10:40:31.628 [INF] WLLT: Returning address (account=0 branch=1 child=96)
2020-04-28 10:40:31.628 [INF] WLLT: Returning address (account=0 branch=1 child=97)
2020-04-28 10:40:31.629 [INF] WLLT: Returning address (account=0 branch=1 child=98)
2020-04-28 10:40:31.629 [INF] WLLT: Returning address (account=0 branch=1 child=99)
2020-04-28 10:40:31.629 [INF] WLLT: Returning address (account=0 branch=1 child=100)
2020-04-28 10:40:31.629 [INF] WLLT: Inserting unconfirmed transaction 6d3733876414e143f6a33ffd7a97c990fd24c897ed8c89a96d461171d5f78309
2020-04-28 10:40:31.632 [INF] WLLT: Inserting unconfirmed transaction d75c2c1e44ae6bc5d51895c6f65fca240805f26b49eb27ceca4eb30aa8262957
2020-04-28 10:40:31.633 [INF] RPCS: RPC method getbalance invoked by 127.0.0.1:49860
panic: runtime error: slice bounds out of range [128:0]

goroutine 414105 [running]:
decred.org/dcrwallet/wallet/udb.extractBlockHeaderHeight(...)
	/home/jholdstock/code/decred/dcrwallet/wallet/udb/txmined.go:488
decred.org/dcrwallet/wallet/udb.(*Store).AddCredit(0xc0007e4020, 0xfb54a0, 0xc0003792c0, 0xc0000c0be0, 0xc00071c730, 0x100000001, 0xc000000000, 0x0, 0x0)
	/home/jholdstock/code/decred/dcrwallet/wallet/udb/txmined.go:1293 +0x6dd
decred.org/dcrwallet/wallet.(*Wallet).processTransactionRecord(0xc00013e160, 0xfac0c0, 0xc001dba000, 0xfb06a0, 0xc0013a0268, 0xc0000c0be0, 0x0, 0xc00071c730, 0x0, 0x0, ...)
	/home/jholdstock/code/decred/dcrwallet/wallet/chainntfns.go:667 +0x1f89
decred.org/dcrwallet/wallet.(*Wallet).AddTransaction.func1(0xfb06a0, 0xc0013a0268, 0xd17f52, 0xe)
	/home/jholdstock/code/decred/dcrwallet/wallet/chainntfns.go:350 +0x1bf
decred.org/dcrwallet/wallet/walletdb.Update(0xfac0c0, 0xc001dba000, 0xfac2c0, 0xc00016c600, 0xc000bd7d78, 0x0, 0x0)
	/home/jholdstock/code/decred/dcrwallet/wallet/walletdb/interface.go:249 +0x1a5
decred.org/dcrwallet/wallet.(*Wallet).AddTransaction(0xc00013e160, 0xfac0c0, 0xc001dba000, 0xc000440000, 0x0, 0x0, 0xc00043e000)
	/home/jholdstock/code/decred/dcrwallet/wallet/chainntfns.go:312 +0x1b6
decred.org/dcrwallet/chain.(*Syncer).relevantTxAccepted(0xc0001683f0, 0xfac0c0, 0xc001dba000, 0xc000330c00, 0x1fc, 0x200, 0xc00043e000, 0xbbf6a0)
	/home/jholdstock/code/decred/dcrwallet/chain/sync.go:652 +0x98
decred.org/dcrwallet/chain.(*notifier).Notify(0xc000111f50, 0xc0003284e0, 0x12, 0xc000330c00, 0x1fc, 0x200, 0x0, 0x0)
	/home/jholdstock/code/decred/dcrwallet/chain/sync.go:559 +0x7aa
github.com/jrick/wsrpc/v2.(*Client).in.func2(0xc0006a4100, 0xc0001d8020, 0xc00071c550, 0xfac0c0, 0xc000110270, 0xc000042b40, 0xc00016a600)
	/home/jholdstock/code/gopath/pkg/mod/github.com/jrick/wsrpc/v2@v2.3.3/rpc.go:388 +0x1a3
created by github.com/jrick/wsrpc/v2.(*Client).in
	/home/jholdstock/code/gopath/pkg/mod/github.com/jrick/wsrpc/v2@v2.3.3/rpc.go:381 +0x3df
```

--- a/wallet/chainntfns.go
+++ b/wallet/chainntfns.go
@@ -330,7 +330,7 @@ func (w *Wallet) AddTransaction(ctx context.Context, tx *wire.MsgTx, blockHash *
 		}
 
 		var header *wire.BlockHeader
-		var meta udb.BlockMeta
+		var meta *udb.BlockMeta
 		switch {
 		case blockHash != nil:
 			inChain, _ := w.txStore.BlockInMainChain(dbtx, blockHash)
@@ -341,13 +341,14 @@ func (w *Wallet) AddTransaction(ctx context.Context, tx *wire.MsgTx, blockHash *
 			if err != nil {
 				return err
 			}
-			meta, err = w.txStore.GetBlockMetaForHash(txmgrNs, blockHash)
+			meta = new(udb.BlockMeta)
+			*meta, err = w.txStore.GetBlockMetaForHash(txmgrNs, blockHash)
 			if err != nil {
 				return err
 			}
 		}
 
-		watchOutPoints, err = w.processTransactionRecord(ctx, dbtx, rec, header, &meta)
+		watchOutPoints, err = w.processTransactionRecord(ctx, dbtx, rec, header, meta)
 		return err
 	})
 	if err != nil {

